Identifier of "RegisteredServiceImpl_Properties" is too long for Oracle

When starting up CAS 4.2.0 with a JPA service registry handler on Oracle 12.1c, Hibernate is unable to update the schema with:

```
Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Unable to execute schema management to JDBC target [create table RegisteredServiceImpl_Properties (AbstractRegisteredService_id number(19,0) not null, properties_id number(19,0) not null, properties_KEY varchar2(255 char) not null, primary key (AbstractRegisteredService_id, properties_KEY))]
	at org.hibernate.tool.schema.internal.TargetDatabaseImpl.accept(TargetDatabaseImpl.java:59)
	at org.hibernate.tool.schema.internal.SchemaMigratorImpl.applySqlString(SchemaMigratorImpl.java:421)
	at org.hibernate.tool.schema.internal.SchemaMigratorImpl.applySqlStrings(SchemaMigratorImpl.java:410)
	at org.hibernate.tool.schema.internal.SchemaMigratorImpl.createTable(SchemaMigratorImpl.java:226)
	at org.hibernate.tool.schema.internal.SchemaMigratorImpl.doMigrationToTargets(SchemaMigratorImpl.java:167)
	at org.hibernate.tool.schema.internal.SchemaMigratorImpl.doMigration(SchemaMigratorImpl.java:60)
	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:129)
	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:97)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:481)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:444)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:802)
	... 63 more
Caused by: java.sql.SQLSyntaxErrorException: ORA-00972: identifier is too long

	at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:450)
	at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:399)
...
```

This is running in Tomcat 8 with the following relevant `cas.properties` settings:

```
svcreg.database.dialect=org.hibernate.dialect.Oracle12cDialect
svcreg.database.batchSize=20
svcreg.database.ddl.auto=update
```

And `deployerConfigContext.xml` snippet:

```xml
    <bean id="dataSourceService" class="org.springframework.jndi.JndiObjectFactoryBean"
          p:jndiName="java:comp/env/jdbc/ou_cas"/>

    <alias name="jpaServiceRegistryDao" alias="serviceRegistryDao" />
```

Note that, `web.xml` was modified to load the customized XML files last:

```xml
    <context-param>
        <param-name>contextConfigLocation</param-name>
        <param-value>
            classpath*:/META-INF/spring/*.xml
            /WEB-INF/spring-configuration/*.xml
            /WEB-INF/deployerConfigContext.xml
        </param-value>
    </context-param>
```

Steps to reproduce:
1. Setup an Oracle 12 database, XE works fine. Docker containers work great.
2. Create a schema.
3. Setup CAS with a JPA Service Registry.

--- a/cas-server-core-services/src/main/java/org/jasig/cas/services/AbstractRegisteredService.java
+++ b/cas-server-core-services/src/main/java/org/jasig/cas/services/AbstractRegisteredService.java
@@ -123,7 +123,7 @@ public abstract class AbstractRegisteredService implements RegisteredService, Co
     private RegisteredServicePublicKey publicKey;
 
     @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
-    @JoinTable(name="RegisteredServiceImpl_Properties")
+    @JoinTable(name="RegisteredServiceImpl_Props")
     private Map<String, DefaultRegisteredServiceProperty> properties = new HashMap<>();
 
     @Override

